sem/utils.py

```diff
@@ -540,57 +544,62 @@ def parse_logs(log_file):
         log_file (string): Path to where the log file is stored
     """
     log_list = []
+    regex = re.compile(r'[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (\[.*?\] )?(\w+):(\w+)\((.*?)\)(: \[(\w+)\s*\] (.*))?')
```

Comment:
I agree with you. But after our last call, I searched online and it seems that having more groups slows down Regex matching and hence I did not add a new group only to remove the outer square brackets. But if we want to do this we can do so easily by using the following regex:
`regex = re.compile(r'[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (\[**(.*?)**\] )?(\w+):(\w+)\((.*?)\)(: \[(\w+)\s*\] (.*))?')`

> same for the last part.

This last part space is not captured in any group. I do not think we need to change this.

Reply:
> I agree with you. But after our last call, I searched online and it seems that having more groups slows down Regex matching and hence I did not add a new group only to remove the outer square brackets. But if we want to do this we can do so easily by using the following regex:
> `regex = re.compile(r'[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (\[(.*?)\] )?(\w+):(\w+)\((.*?)\)(: \[(\w+)\s*\] (.*))?')`

So you suggest to avoid using a non-capturing group? From [here](https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups) it looks like there's no difference in performance from one to the other:
_"It should be mentioned that there’s no performance difference in searching between capturing and non-capturing groups; neither form is any faster than the other."_

> This last part space is not captured in any group. I do not think we need to change this.

Here I was basically proposing to use a non-capturing group since we do not need it

Reply:
I misunderstood capturing and non-capturing groups (This is what I referred to [here](https://stackoverflow.com/questions/41444807/why-is-regex-search-slower-with-capturing-groups-in-python#:~:text=When%20timing%20performance%20of%20two,operations%20internally%20in%20regex%20module.)). I completely agree with your proposed regex now.